state_client.go

```diff
@@ -174,6 +181,9 @@ func unmarshalWalletState(rpcState lnrpc.WalletState) (WalletState, error) {
 	case lnrpc.WalletState_RPC_ACTIVE:
 		return WalletStateRPCActive, nil
 
+	case lnrpc.WalletState_SERVER_ACTIVE:
```

Comment:
Perhaps we should just ignore all unknown states here? As otherwise, we need to update this each time we add a new state.

Reply:
I thought about this a bit and don't think it's a good idea. Because those new states likely mean something has changed in the business logic. And we also possibly need to implement some backward compatible logic here anyway. For example I needed to add the `ReadyForGetInfo()` method to make sure this works with both 0.13.x and 0.14.x nodes. So I think it's good to error out on unknown states.